### PR TITLE
RIA-8048 fee_support_screens

### PR DESCRIPTION
**Pull request by ikirsanov**
### Jira link (if applicable)
https://tools.hmcts.net/jira/browse/RIA-8048

### Change description ###
Added FeeSupport screens journey, controllers, views, translations, tests.

### Checklist

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: app/controllers/appeal-application/asylum-support.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+import LaunchDarklyService from '../../service/launchDarkly-service';
+import UpdateAppealService from '../../service/update-appeal-service';
+import { shouldValidateWhenSaveForLater } from '../../utils/save-for-later-utils';
+import { getConditionalRedirectUrl } from '../../utils/url-utils';
+import { getRedirectPage } from '../../utils/utils';
+import { asylumSupportValidation } from '../../utils/validations/fields-validations';
+
+async function getAsylumSupport(req: Request, res: Response, next: NextFunction) {
+  try {
+    const drlmSetAsideFlag = await LaunchDarklyService.getInstance().getVariation(req, FEATURE_FLAGS.DLRM_FEE_REMISSION_FEATURE_FLAG, false);
```
[the remark this answers is not in the document]
Comment (ikirsanov) [reply app/controllers/appeal-application/asylum-support.ts]: renamed in all components

File: app/controllers/appeal-application/fee-waiver.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    });
+  } catch (error) {
+    next(error);
+  }
+}
+
+function postFeeWaiver(updateAppealService: UpdateAppealService) {
+  return async (req: Request, res: Response, next: NextFunction) => {
+    async function persistAppeal(appeal: Appeal, drlmSetAsideFlag) {
+      const appealUpdated: Appeal = await updateAppealService.submitEventRefactored(Events.EDIT_APPEAL, appeal, req.idam.userDetails.uid, req.cookies['__auth-token'], drlmSetAsideFlag);
```
[the remark this answers is not in the document]
Comment (ikirsanov) [reply app/controllers/appeal-application/fee-waiver.ts:30]: Moved these logic in all Post functions to the beginning of the function to check it first.

File: app/controllers/appeal-application/help-with-fees-reference-number.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+import LaunchDarklyService from '../../service/launchDarkly-service';
+import UpdateAppealService from '../../service/update-appeal-service';
+import { shouldValidateWhenSaveForLater } from '../../utils/save-for-later-utils';
+import { getConditionalRedirectUrl } from '../../utils/url-utils';
+import { getRedirectPage } from '../../utils/utils';
+import { helpWithFeesRefNumberValidation } from '../../utils/validations/fields-validations';
+
+async function getHelpWithFeesRefNumber(req: Request, res: Response, next: NextFunction) {
+  try {
+    const drlmSetAsideFlag = await LaunchDarklyService.getInstance().getVariation(req, FEATURE_FLAGS.DLRM_FEE_REMISSION_FEATURE_FLAG, false);
```
[the remark this answers is not in the document]
Comment (ikirsanov) [reply app/controllers/appeal-application/help-with-fees-reference-number.ts]: ALready changed everywhere

File: app/service/update-appeal-service.ts [hunk 4]
```diff
@@ -650,6 +656,49 @@ export default class UpdateAppealService {
         caseData.appealType = appeal.application.appealType;
       }
 
+      if (appeal.application.remissionOption) {
+        caseData.remissionOption = appeal.application.remissionOption;
```
[the remark this answers is not in the document]
Comment (ikirsanov) [reply app/service/update-appeal-service.ts:660]: We already discussed that we will have new one

File: app/utils/payments-utils.ts [hunk 5]
```diff
@@ -4,6 +4,7 @@ import { States } from '../data/states';
 const EA_HU_EUSS_APPEAL_TYPES = ['refusalOfHumanRights', 'refusalOfEu', 'euSettlementScheme'];
 
 export function getFee(appeal: Appeal) {
+  //  TODO if selected Deprivation - it could throw an error because feeWithHearing and feeWithoutHearing will be nulls!
```
[the remark this answers is not in the document]
Comment (ikirsanov) [reply app/utils/payments-utils.ts:7]: I will clarify with Elli and we will create this ticket

**Review by MiguelMartinRojas — APPROVED** [review]

**Review by colmlynch60 — APPROVED** [review]
Quite a large PR but LGTM

**Pull request merged**